bcipy/signal/model/offline_analysis.py

```diff
@@ -528,31 +474,42 @@ def offline_analysis(
     assert len(data_file_paths) < 3, "BciPy only supports up to 2 devices for offline analysis."
     assert len(data_file_paths) > 0, "No data files found for offline analysis."
 
-    models = []
-    log.info(f"Starting offline analysis for {data_file_paths}")
-    for raw_data_path in data_file_paths:
-        raw_data = load_raw_data(raw_data_path)
-        device_spec = devices_by_name.get(raw_data.daq_type)
-        # extract relevant information from raw data object eeg
-
-        if device_spec.content_type == "EEG" and device_spec.is_active:
-            erp_model = analyze_erp(
-                raw_data,
-                parameters,
-                device_spec,
-                data_folder,
-                estimate_balanced_acc,
-                save_figures,
-                show_figures)
-            models.append(erp_model)
-
-        if device_spec.content_type == "Eyetracker" and device_spec.is_active:
-            et_model = analyze_gaze(
-                raw_data, parameters, device_spec, data_folder, save_figures, show_figures, model_type="Individual")
-            models.append(et_model)
-
-    if len(models) > 1:
-        log.info("Multiple Models Trained. Fusion Analysis Not Yet Implemented.")
+    if len(data_file_paths) == 2:
+        # Note that the models trained here are not saved/used for the online system.
+        log.info("Multiple Devices Found. Starting fusion analysis...")
+        eeg_data = load_raw_data(data_file_paths[0])
+        device_spec_eeg = devices_by_name.get(eeg_data.daq_type)
+        gaze_data = load_raw_data(data_file_paths[1])
+        device_spec_gaze = devices_by_name.get(gaze_data.daq_type)
+        symbol_set = alphabet()
+        eeg_acc, gaze_acc, fusion_acc = calculate_eeg_gaze_fusion_acc(
+            eeg_data, gaze_data, device_spec_eeg, device_spec_gaze, symbol_set, parameters, data_folder)
+        log.info(f"EEG Accuracy: {eeg_acc}, Gaze Accuracy: {gaze_acc}, Fusion Accuracy: {fusion_acc}")
+
+    # Ask the user if they want to proceed with full dataset model training
+    if confirm("Would you like to proceed with model training?"):
+        models = []
+        log.info(f"Starting offline analysis for {data_file_paths}")
```

Comment:
Non-blocking comment: We could add the data or device types in the confirmation message!

Reply:
Also non-blocking, but it seems like functionality regarding interaction (confirmation) should be separated out from the functionality of training a model.